hk2-locator/src/test/java/org/glassfish/hk2/tests/locator/negative/classanalysis/NegativeClassAnalysisModule.java

```diff
@@ -23,22 +23,23 @@
 
 /**
  * @author jwells
- *
  */
 public class NegativeClassAnalysisModule implements TestModule {
 
-    /* (non-Javadoc)
-     * @see org.glassfish.hk2.tests.locator.utilities.TestModule#configure(org.glassfish.hk2.api.DynamicConfiguration)
-     */
+    private final String analyzer;
+
+    public NegativeClassAnalysisModule(final String analyzerName) {
+        this.analyzer = analyzerName;
+    }
+
     @Override
     public void configure(DynamicConfiguration config) {
         config.addActiveDescriptor(ConfigurablyBadClassAnalyzer.class);
-        
+
         config.bind(BuilderHelper.link(SelfAnalyzer.class.getName()).
                 to(ClassAnalyzer.class.getName()).
-                analyzeWith(NegativeClassAnalysisTest.SELF_ANALYZER).
+                analyzeWith(this.analyzer).
```

Comment:
Why was this changed?

Reply:
Good question ... I had to think about it again.

Previous version: used (SELF_ANALYZER=) Narcis analyzer to analyze Narcis service (forbidden but exception was ignored). Then locator was used with it's parent in test.
The last test was targeting forbidden cyclic usage of the analyzer to reify itself. 

Current version: uses default analyzer to analyze Narcis service without ignoring exceptions. Then all tests except final two use "BadConfigurableAnalyzer" to test several error scenarios. The final two tests (one was added) target forbidden cyclic dependencies - the longer test in effect tests the original scenario with ignorance and disabling it. 
The shorter test is added and tests fast fail if there is cyclic dependency at the time of ServiceLocator creation.